svc.go

```diff
@@ -167,21 +171,83 @@ func (as *autoNATService) doDial(pi peer.AddrInfo) *pb.Message_DialResponse {
 	ctx, cancel := context.WithTimeout(as.ctx, as.config.dialTimeout)
 	defer cancel()
 
-	as.config.dialer.Peerstore().ClearAddrs(pi.ID)
+	// START DIALING
+	// 1. Aim is to diversify dial attempts across address groups where the grouping key is (IP address + transport protocol(port dosen't matter))
+	// 2. We will continue dialing addresses in a group till we see a successful dial for an address.
```

Comment:
The server should have a bounded number of dial attempts it makes for a response. that shouldn't extend the quota of work it's willing to do for a client just because some of the addresses are failing.